# Olympy

In [54]:
# installing pandas package
!pip install pandas

In [55]:
import pandas as pd

In [56]:
# check current installed pandas version
pd.__version__

'2.2.2'

In [57]:
# Colab pandas doc access
pd.read_csv?

Pandas doc: https://pandas.pydata.org/docs/user_guide/index.html#user-guide

In [58]:
# downloading the dataset
from pathlib import Path

if not Path('dataset.csv').exists():
  !wget https://raw.githubusercontent.com/valdal14/olympy/refs/heads/main/dataset.csv

If you prefer to upload your own dataset, please uncommet the following code block and upload your own.

In [59]:
# from google.colab import files
# files.upload()

In [60]:
# check the dataset upload
!ls -la

total 2332
drwxr-xr-x 1 root root    4096 Feb 18 07:29 .
drwxr-xr-x 1 root root    4096 Feb 18 07:27 ..
drwxr-xr-x 4 root root    4096 Jan 16 14:24 .config
-rw-r--r-- 1 root root 2371208 Feb 18 07:29 dataset.csv
drwxr-xr-x 1 root root    4096 Jan 16 14:24 sample_data


In [61]:
# read the flat csv file using pandas
# Skiping the first 5 rows due to the dataset file description and disclaimer
olympic_data = pd.read_csv('dataset.csv', skiprows=5)

In [62]:
# check the head to verify the file was uploaded and preview the first 3 rows
olympic_data.head(3)


Year    City      Sport     Discipline        Athlete Name  NOC Gender  \
0  1896  Athens    Cycling  Cycling Track       FLAMENG, Léon  FRA    Men   
1  1896  Athens    Cycling  Cycling Track  KOLETTIS, Georgios  GRE    Men   
2  1896  Athens  Athletics      Athletics       LANE, Francis  USA    Men   

   Event Event Gender   Medal  Position  
0  100km            M    Gold         1  
1  100km            M  Silver         2  
2   100m            M  Bronze         3

# Pandas Attributes & Methods

In [63]:
# .shape - Returns the rows and columns
olympic_data.shape?

olympic_data.shape

(27174, 11)

In [64]:
# .tail() - Returns the last n rows
olympic_data.tail?

olympic_data.tail(3)

Year    City    Sport Discipline       Athlete Name  NOC Gender  \
27171  2004  Athens  Fencing    Fencing     BOKEL, Claudia  GER  Women   
27172  2004  Athens  Fencing    Fencing    DUPLITZER, Imke  GER  Women   
27173  2004  Athens  Fencing    Fencing  HEIDEMANN, Britta  GER  Women   

           Event Event Gender   Medal  Position  
27171  épée team            W  Silver         2  
27172  épée team            W  Silver         2  
27173  épée team            W  Silver         2

In [65]:
# .sample() - Returns a random sample of items from an axis of object
olympic_data.sample?

olympic_data.sample(3)

Year     City          Sport     Discipline          Athlete Name  NOC  \
15618  1980   Moscow  Weightlifting  Weightlifting         NIKITIN, Igor  URS   
22243  1996  Atlanta         Rowing         Rowing  PAVLOVICH, Yaroslava  BLR   
18083  1988    Seoul  Weightlifting  Weightlifting     ZAKHAREVICH, Yuri  URS   

      Gender                                  Event Event Gender   Medal  \
15618    Men  90 - 100kg, total (first-heavyweight)            M  Silver   
22243  Women               eight with coxswain (8+)            W  Bronze   
18083    Men       100 - 110kg, total (heavyweight)            M    Gold   

       Position  
15618         2  
22243         3  
18083         1

In [66]:
# .info() - Print a concise summary including the types (dtype) of a DataFrame
olympic_data.info?

olympic_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 27174 entries, 0 to 27173
Data columns (total 11 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   Year          27174 non-null  int64 
 1   City          27174 non-null  object
 2   Sport         27174 non-null  object
 3   Discipline    27174 non-null  object
 4   Athlete Name  27174 non-null  object
 5   NOC           27174 non-null  object
 6   Gender        27174 non-null  object
 7   Event         27174 non-null  object
 8   Event Gender  27174 non-null  object
 9   Medal         27174 non-null  object
 10  Position      27174 non-null  int64 
dtypes: int64(2), object(9)
memory usage: 2.3+ MB


In [67]:
# .describe() - Generate descriptive statistics.
olympic_data.describe?
# Only computed for entities that holds numeric values in the dataset
olympic_data.describe()

Year      Position
count  27174.000000  27174.000000
mean    1964.685803      1.992566
std       31.590396      0.817469
min     1896.000000      1.000000
25%     1936.000000      1.000000
50%     1972.000000      2.000000
75%     1992.000000      3.000000
max     2004.000000      3.000000

# Pandas Series
Pandas allows us to select a columns or view more columns. When we import a file using one of pandas read method, the type is a DataFrame. We can use the `type()` method to confirm it. However, we can select the columns we want or need to work with.

In [68]:
type(olympic_data)

pandas.core.frame.DataFrame

In [69]:
# select a single column. Rememeber it is case sensitive
athlete_col = olympic_data['Athlete Name']
athlete_col

0             FLAMENG, Léon
1        KOLETTIS, Georgios
2             LANE, Francis
3          SZOKOLYI, Alajos
4             BURKE, Thomas
                ...        
27169    LOGOUNOVA, Tatiana
27170         SIVKOVA, Anna
27171        BOKEL, Claudia
27172       DUPLITZER, Imke
27173     HEIDEMANN, Britta
Name: Athlete Name, Length: 27174, dtype: object

In [70]:
# using the .columns attributed we can see a list of all columns
olympic_data.columns

Index(['Year', 'City', 'Sport', 'Discipline', 'Athlete Name', 'NOC', 'Gender',
       'Event', 'Event Gender', 'Medal', 'Position'],
      dtype='object')

In [71]:
# Select a single column and check its type
type(athlete_col)

pandas.core.series.Series

In [72]:
# We can also use the dot notations since columns are treated as attributes
# This notation works only if there are no spaces in the column I want to check
olympic_data.Sport

0          Cycling
1          Cycling
2        Athletics
3        Athletics
4        Athletics
           ...    
27169      Fencing
27170      Fencing
27171      Fencing
27172      Fencing
27173      Fencing
Name: Sport, Length: 27174, dtype: object

In [73]:
# If there are spaces, we need to use the [] notation
olympic_data['Athlete Name']

0             FLAMENG, Léon
1        KOLETTIS, Georgios
2             LANE, Francis
3          SZOKOLYI, Alajos
4             BURKE, Thomas
                ...        
27169    LOGOUNOVA, Tatiana
27170         SIVKOVA, Anna
27171        BOKEL, Claudia
27172       DUPLITZER, Imke
27173     HEIDEMANN, Britta
Name: Athlete Name, Length: 27174, dtype: object

In [74]:
# We can then use the columns properties to check additional information about what's stored
# Return unique values of Series object.
olympic_data.Sport.unique?

olympic_data.Sport.unique()

array(['Cycling', 'Athletics', 'Aquatics', 'Shooting', 'Tennis',
       'Fencing', 'Weightlifting', 'Gymnastics', 'Wrestling', 'Sailing',
       'Archery', 'Basque Pelota', 'Cricket', 'Croquet', 'Rowing',
       'Football', 'Equestrian', 'Golf', 'Polo', 'Rugby', 'Tug of War',
       'Boxing', 'Roque', 'Lacrosse', 'Water Motorsports', 'Rackets',
       'Hockey', 'Jeu de paume', 'Skating', 'Modern Pentathlon',
       'Ice Hockey', 'Canoe / Kayak', 'Basketball', 'Handball', 'Judo',
       'Volleyball', 'Table Tennis', 'Baseball', 'Badminton', 'Softball',
       'Taekwondo', 'Triathlon'], dtype=object)

In [75]:
# Return a Series containing counts of unique values
olympic_data.Sport.value_counts?

olympic_data.Sport.value_counts()

Sport
Aquatics             3481
Athletics            3271
Rowing               2379
Gymnastics           2115
Fencing              1485
Football             1279
Hockey               1227
Wrestling            1069
Shooting             1060
Sailing              1007
Cycling               954
Canoe / Kayak         918
Basketball            868
Equestrian            849
Volleyball            826
Handball              801
Boxing                798
Weightlifting         503
Judo                  379
Archery               281
Baseball              263
Tennis                254
Rugby                 192
Modern Pentathlon     168
Softball              135
Table Tennis           96
Badminton              96
Tug of War             94
Polo                   66
Lacrosse               59
Taekwondo              48
Golf                   30
Skating                27
Ice Hockey             27
Cricket                24
Triathlon              12
Rackets                10
Croquet                 8
Water Motorsports       5
Basque Pelota           4
Roque                   3
Jeu de paume            3
Name: count, dtype: int64

# Dataset First Data Analysis

Find the time or data range of the olympic dataset

In [77]:
olympic_data.Year.max()

2004

In [78]:
olympic_data.Year.min()

1896

The range is from 1896 to 2004

The olympic games occur every 4 years, is there any missing data in the dataset?

In [84]:
years = olympic_data.Year.unique()
years

array([1896, 1900, 1904, 1908, 1912, 1920, 1924, 1928, 1932, 1936, 1948,
       1952, 1956, 1960, 1964, 1968, 1972, 1976, 1980, 1984, 1988, 1992,
       1996, 2000, 2004])

In [89]:
missing_editions = [year + 4 for year in years[:-1] if (year + 4) not in years]

In [90]:
# Let's find out if there were a missing edition
missing_editions

[np.int64(1916), np.int64(1940)]

Find out the type of medals awwarded

In [91]:
medals = olympic_data.Medal.unique()
medals

array(['Gold', 'Silver', 'Bronze'], dtype=object)

How many medal type for each type were awarded?

In [98]:
olympic_data.Medal.value_counts()

Medal
Gold      9181
Silver    9014
Bronze    8979
Name: count, dtype: int64

What are the diffrent NOC (National Olympic Committees)?

In [99]:
olympic_data.NOC.unique()

array(['FRA', 'GRE', 'USA', 'HUN', 'GER', 'AUT', 'GBR', 'AUS', 'DEN',
       'ZZX', 'SUI', 'IND', 'NED', 'CAN', 'NOR', 'BEL', 'ESP', 'BOH',
       'ITA', 'SWE', 'CUB', 'RU1', 'FIN', 'RSA', 'ANZ', 'LUX', 'EST',
       'BRA', 'JPN', 'TCH', 'NZL', 'ARG', 'HAI', 'POL', 'URU', 'YUG',
       'ROU', 'POR', 'EGY', 'PHI', 'IRL', 'CHI', 'MEX', 'LAT', 'TUR',
       'KOR', 'PAN', 'JAM', 'SRI', 'PER', 'PUR', 'IRI', 'TRI', 'URS',
       'LIB', 'BUL', 'VEN', 'EUA', 'PAK', 'ISL', 'BAH', 'BWI', 'GHA',
       'IRQ', 'SIN', 'TPE', 'ETH', 'MAR', 'TUN', 'NGR', 'KEN', 'FRG',
       'MGL', 'GDR', 'UGA', 'CMR', 'PRK', 'COL', 'NIG', 'BER', 'THA',
       'TAN', 'GUY', 'ZIM', 'ZAM', 'CHN', 'CIV', 'DOM', 'ALG', 'SYR',
       'SUR', 'CRC', 'SEN', 'AHO', 'DJI', 'ISV', 'INA', 'EUN', 'NAM',
       'IOP', 'QAT', 'ISR', 'LTU', 'CRO', 'SLO', 'MAS', 'RUS', 'UKR',
       'TGA', 'ARM', 'BLR', 'MDA', 'ECU', 'KAZ', 'AZE', 'BDI', 'SVK',
       'CZE', 'UZB', 'GEO', 'MOZ', 'HKG', 'KGZ', 'BAR', 'KSA', 'VIE',
       'MKD', 'KUW',

What's the NOC ZZX is?

In [104]:
zzx_data = olympic_data[olympic_data.NOC == "ZZX"]
zzx_data.sample(3)

Year      City      Sport Discipline           Athlete Name  NOC Gender  \
749  1904  St Louis  Athletics  Athletics          HATCH, Sidney  ZZX    Men   
522  1900     Paris       Polo       Polo  KEENE, Foxhall Parker  ZZX    Men   
750  1904  St Louis  Athletics  Athletics           HEARN, Lacey  ZZX    Men   

           Event Event Gender   Medal  Position  
749  4miles team            M  Silver         2  
522         polo            M    Gold         1  
750  4miles team            M  Silver         2